sources/packages/backend/libs/integrations/src/services/msfaa-number/msfaa-number.service.ts

```diff
@@ -108,10 +113,38 @@ export class MSFAANumberService extends RecordDataModelService<MSFAANumber> {
         "Not all required fields to update a received MSFAA record were provided.",
       );
     }
-
+    // Check if the msfaa number needs to be reactivated or simply updated with a signed date.
+    const retrievedMSFAARecord = await this.repo.findOne({
+      select: {
+        id: true,
+        studentId: true,
+        referenceApplicationId: true,
+        cancelledDate: true,
+        dateSigned: true,
+        offeringIntensity: true,
+      },
+      where: {
+        msfaaNumber,
+      },
+    });
+    if (
+      !retrievedMSFAARecord.dateSigned &&
+      retrievedMSFAARecord.cancelledDate
+    ) {
+      // This msfaa number was originally cancelled and is being reactivated now. Re-associate all disbursements pending e-cert generation for the same offering intensity with this msfaa number.
+      await this.msfaaNumberSharedService.internalCreateMSFAANumber(
+        retrievedMSFAARecord.studentId,
+        retrievedMSFAARecord.referenceApplicationId,
+        retrievedMSFAARecord.offeringIntensity,
+        (
+          await this.systemUsersService.systemUser()
+        ).id,
+        { msfaaNumber: { id: retrievedMSFAARecord.id } as MSFAANumber },
+      );
+    }
```

Comment:
Should the `update` not share the same DB transaction from `internalCreateMSFAANumber`?